pytorch_lightning/callbacks/intel_compressor.py

```diff
@@ -0,0 +1,116 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Intel Quantization (TODO: pruning)
+^^^^^^^^^^^^
+
+"""
+from enum import Enum
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core import LightningDataModule
+from pytorch_lightning.trainer.states import TrainerFn, TrainerStatus
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class QuantizationMode(Enum):
+    PTQ_DYNAMIC = "post_training_dynamic_quant"
+    PTQ_STATIC = "post_training_static_quant"
+    QAT = "quant_aware_training"
+
+
+class INCQuantization(Callback):
+    """Quantization allows speeding up inference and decreasing memory requirements by performing computations and
+    storing tensors at lower bitwidths (such as INT8) than floating point precision. And this callback will
+    quantized model with Intel Neural Compressor(INC) tool.
+
+    Args:
+
+        config_path_or_obj: config file or Quantization_Conf class for INC.
+
+        monitor: Specified metric which will be computed in evaluation function.
+
+        module_name_to_quant: the model name which you want to quantize, it should be torch.nn.Module.
+
+        dirpath: save path where the quantization config and weights be saved to.
+
+    .. Intel Neural Compressor: https://github.com/intel/neural-compressor
+    """
+
+    def __init__(
+        self,
+        config_path_or_obj,
+        monitor: str,
+        module_name_to_quant: str,
+        datamodule: LightningDataModule = None,
+        dirpath: str = None,
+    ) -> None:
+        try:
+            from neural_compressor.conf.config import Quantization_Conf
+        except ImportError as e:
+            raise RuntimeError("Please install neural-compressor with: pip install neural-compressor") from e
+
+        self.config = (
+            config_path_or_obj
+            if isinstance(config_path_or_obj, Quantization_Conf)
+            else Quantization_Conf(config_path_or_obj)
+        )
+        self.approach = self.config.usr_cfg.quantization.approach
+        assert monitor is not None, "Please set metric name for evaluation loop!"
+        self.monitor = monitor
+        assert (
+            module_name_to_quant is not None
+        ), "Please set module_name_to_quant with module name which you want to quantize"
+        self.module_name_to_quant = module_name_to_quant
+        self.datamodule = datamodule
+        self.dirpath = dirpath
+
+    def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        """Called when the compressor loop begins."""
+        try:

```

Comment:
Use our imports mechanism to check this. This is cleaner.

Reply:
done